### PR TITLE
Adds support for GCE credentials in credential form(s)

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
Small portion of #5879 

The GCE subform has 4 (really 3) fields:

Screen Shot 2020-04-14 at 10 43 28 AM

1. Service account JSON file: This isn't actually a field that we POST to the credentials endpoint, it's more of a shortcut to allow users to fill out the rest of the fields without having to copy/paste.  A user can upload their service account file and we'll parse and populate the relevant fields.
2. Service account email address: Required text field.  This field wasn't restricted by format (email) in the old app so I followed suit here.  This field actually maps to `username` when we POST so that may be why there's no format checking.
3. Project: Optional text field
4. RSA private key: Required textarea.  This is analogous to the SSH private key fields on other credential types.  

The last 3 fields themselves are not very interesting.  Almost all of this work went into the first field.  Here are a few notes about that first field (JSON file selector)

1. Only .json files are supported.  This is restricted by the dropzone lib that PF is using under the hood.  If a user drops a different file type on the field an error should be shown.
2. If a user selects or drops more than one file on the field then an error should be shown.
3. If the user selects a .json file with syntax errors (we can't parse) then an error should be shown.
4. If the user selects a .json file without any of the expected fields (client_email, project_id, private_key) then an error should be shown.
5. If the user selects a .json file with at least one of the expected fields then we clear out the current values for all three and then add values for the present expected fields from the file.
6. If the user uploads a file and then "clears" it we clear out all three fields.

Example .json file can be found here: https://github.com/ansible/awx/blob/devel/awx/ui/test/e2e/tests/gce.json

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Credential/shared/CredentialForm.test.jsx [hunk 1]
```diff
@@ -313,36 +413,71 @@ const credentialTypes = [
...
+          multiline: true,
+          help_text:
+            'Paste the contents of the PEM file associated with the service account email.',
+        },
+      ],
+      required: ['username', 'ssh_key_data'],
+    },
+    injectors: {},
+  },
 ];
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Credential/shared/CredentialForm.test.jsx]: this file is getting a bit big.  if we move all these at the top to data files, I think we'd be able to get this test file down to ~300 lines, which might be a bit better.

**Review by jlmitch5 — COMMENTED** [review]
looks good,  only suggestion I have is to break those data objects to data files for the test

File: awx/ui_next/src/screens/Credential/shared/CredentialForm.jsx [hunk 2]
```diff
@@ -103,6 +108,7 @@ function CredentialFormFields({
             <Title size="md">{i18n._(t`Type Details`)}</Title>
             {
               {
+                [gceCredentialTypeId]: <GoogleComputeEngineSubForm />,
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Credential/shared/CredentialForm.jsx]: on Line 105 you are checking `formik.values.credential_type`.  You could just get the value off of credential_type field(using useField) above on line 30.

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
Things look good to me.  I only have 1 recommendation here regarding the `credential_type` value

File: awx/ui_next/src/screens/Credential/shared/CredentialForm.jsx (hunk 2 above)
Comment (mabashian) [reply awx/ui_next/src/screens/Credential/shared/CredentialForm.jsx]: Awesome - good call.  I think it's definitely cleaner to access the value via the hook.

**Review by AlexSCorey — APPROVED** [review]
Looks good to me.

**Comment by appuk** [comment]
I verified the GCE credential along with different field validation scenarios mentioned in the description and everything works fine! so Looks good to merge 👍

**Pull request merged**